codice/ddf
# [2.26.x] DDF-6565 create a video thumbnail service

#### What does this PR do?
Create a separate video thumbnail service from the existing VideoThumbnailPlugin. Update the plugin to use the new service. A downstream project needs to create video thumbnail outside of the traditional ingest workflow.

NOTE: I was not sure on the best location for the module in the source tree. I'm open to other locations. 

#### Who is reviewing it? 
@jlcsmith 
@derekwilhelm 

#### Select relevant component teams: 
@codice/core-apis 
@codice/data 
@codice/io 
@codice/security  (because I touched a policy file)

#### Ask 2 committers to review/merge the PR and tag them here.
@jrnorth 
@ahoffer


#### How should this be tested?

Ingest a video and verify that a thumbnail was generated. Make sure the size of the test file is smaller than the size limit, which is configured in Video Thumbnail Service.

#### Any background context you want to provide?

#### What are the relevant tickets?
Fixes: #____

#### Screenshots


#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
